src/runtime/plugin.ts

```diff
@@ -5,6 +5,17 @@ export default defineNuxtPlugin(() => {
   const config = useRuntimeConfig();
   const { hostname } = useRequestURL();
 
+  const customDomains = (config.public.customDomains || {}) as Partial<Record<string, string>>;
```

Comment:
I think this partial type is unnecessary here

```suggestion
  const customDomains = (config.public.customDomains || {}) as Record<string, string>;
```